presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -257,7 +263,18 @@ private Date getDate(int columnIndex, DateTimeZone localTimeZone)
         }
 
         try {
-            return new Date(DATE_FORMATTER.withZone(localTimeZone).parseMillis(String.valueOf(value)));
+            long millis = DATE_FORMATTER.withZone(localTimeZone).parseMillis(String.valueOf(value));
+            if (millis >= GREGORIAN_CALENDAR_INTRODUCTION) {
+                return new Date(millis);
+            }
+            // the chronology used by default by Joda is not appropriate for dates preceding the introduction of the Gregorian calendar,
```

Comment:
it is "appropriate" in some way. I think it would be better to indicate the actual problem

```suggestion
            // The chronology used by Joda is not consistent with java.sql.Date
             for dates preceding the introduction of the Gregorian calendar.
            // Same millisecond value represents a different year/month/day.
```

Reply:
I like the "historically accurate" term

Reply:
of course it's not historically accurate for >1583 for some (many) countries as well